interface/forms/newpatient/common.php

```diff
@@ -8,845 +8,42 @@
-                                    <option value='_blank'>-- <?php echo xlt('Select One'); ?> --</option>
-                                    <?php
-                                    //Bring only patient and group categories
-                                    $visitSQL = "SELECT pc_catid, pc_catname, pc_cattype
-                                                   FROM openemr_postcalendar_categories
-                                                   WHERE pc_active = 1 and pc_cattype IN (0,3) and pc_constant_id  != 'no_show' ORDER BY pc_seq";
-
-                                    $visitResult = sqlStatement($visitSQL);
-                                    $therapyGroupCategories = [];
-
-                                    while ($row = sqlFetchArray($visitResult)) {
-                                        $catId = $row['pc_catid'];
-                                        $name = $row['pc_catname'];
-
-                                        if ($row['pc_cattype'] == 3) {
-                                            $therapyGroupCategories[] = $catId;
-                                        }
-
-                                        if ($catId === "_blank") {
-                                            continue;
-                                        }
-
-                                        if ($row['pc_cattype'] == 3 && !$GLOBALS['enable_group_therapy']) {
-                                            continue;
-                                        }
-
-                                        // Fetch acl for category of given encounter. Only if has write auth for a category, then can create an encounter of that category.
-                                        $postCalendarCategoryACO = AclMain::fetchPostCalendarCategoryACO($catId);
-                                        if ($postCalendarCategoryACO) {
-                                            $postCalendarCategoryACO = explode('|', $postCalendarCategoryACO);
-                                            $authPostCalendarCategoryWrite = AclMain::aclCheckCore($postCalendarCategoryACO[0], $postCalendarCategoryACO[1], '', 'write');
-                                        } else { // if no aco is set for category
-                                            $authPostCalendarCategoryWrite = true;
-                                        }
-
-                                        //if no permission for category write, don't show in drop-down
-                                        if (!$authPostCalendarCategoryWrite) {
-                                            continue;
-                                        }
-
-                                        $optionStr = '<option value="%pc_catid%" %selected%>%pc_catname%</option>';
-                                        $optionStr = str_replace("%pc_catid%", attr($catId), $optionStr);
-                                        $optionStr = str_replace("%pc_catname%", text(xl_appt_category($name)), $optionStr);
```

Comment:
There are some specialized translation function (like the one above) that don't appear to be carried over in the twig elements.
(actually, this may be the only one since not seeing any other xl_ function calls that were removed her)

Reply:
I moved this into the controller layer in the C_EncounterVisitForm.php:L218-226
```

            $item = [
                'id' => $row['pc_catid'],
                'name' => xl_appt_category($row['pc_catname']),
                'selected' => ($viewmode && $encounter['pc_catid'] == $row['pc_catid'])
            ];
            if (!$viewmode && $default_visit_category == $row['pc_catid']) {
                $item['selected'] = true;
            }
            $categories[] = $item;
```

Which means the twig just does the following:
```
<option value="{{ category.id|attr }}" {{ category.selected ? 'selected' : '' }}>
                    {{ category.name|text }}
                </option>
```

Reply:
Let me know if I'm missing something.  Otherwise I think I should be able to bring this in.